core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -619,6 +618,29 @@ public PlanOptimizers(
                                 new PruneTableScanColumns(metadata),
                                 new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
 
+        // Try to pushdown aggregations before MultipleDistinctAggregationToMarkDistinct.
+        // This needs to be done after redirections.
+        builder.add(
+                new IterativeOptimizer(
+                        metadata,
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.of(
+                                new PushAggregationIntoTableScan(metadata))),
+                new IterativeOptimizer(
+                        metadata,
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.<Rule<?>>builder()
+                                .addAll(columnPruningRules)
+                                .addAll(projectionPushdownRules)
+                                .add(new PushLimitThroughMarkDistinct())
+                                .add(new MultipleDistinctAggregationToMarkDistinct())
+                                .add(new SingleDistinctAggregationToGroupBy())
```

Comment:
Why is this rule added here?

Reply:
Random noise I guess. It does not seem needed to me. Will remove.